server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -843,4 +847,30 @@ protected Analyzer getWrappedAnalyzer(String fieldName) {
             return defaultAnalyzer;
         }
     }
+
+    public void reloadSearchAnalyzers(AnalysisRegistry registry) throws IOException {
+        logger.info("reloading search analyzers");
+
+        // refresh indexAnalyzers and search analyzers
+        this.indexAnalyzers = registry.rebuildIndexAnalyzers(this.indexAnalyzers, indexSettings);
+        this.searchAnalyzer = new MapperAnalyzerWrapper(this.indexAnalyzers.getDefaultSearchAnalyzer(), p -> p.searchAnalyzer());
+        this.searchQuoteAnalyzer = new MapperAnalyzerWrapper(this.indexAnalyzers.getDefaultSearchQuoteAnalyzer(),
+                p -> p.searchQuoteAnalyzer());
+
+        // also reload search time analyzers in MappedFieldTypes
+        // refresh search time analyzers in MappedFieldTypes
+        List<MappedFieldType> mftsToRefresh = StreamSupport.stream(fieldTypes.spliterator(), false)
+                .filter(mft -> (mft.searchAnalyzer() != null && mft.searchAnalyzer().getAnalysisMode() == AnalysisMode.SEARCH_TIME)
+                        || (mft.searchQuoteAnalyzer() != null && mft.searchQuoteAnalyzer().getAnalysisMode() == AnalysisMode.SEARCH_TIME))
+                .collect(Collectors.toList());
+        List<MappedFieldType> updated = mftsToRefresh.stream().map(mft -> {
+            MappedFieldType newMft = mft.clone();
+            newMft.setSearchAnalyzer(indexAnalyzers.get(mft.searchAnalyzer().name()));
+            newMft.setSearchQuoteAnalyzer(indexAnalyzers.get(mft.searchQuoteAnalyzer().name()));
+            newMft.freeze();
+            return newMft;
+        }).collect(Collectors.toList());
+        fieldTypes = fieldTypes.copyAndAddAll(updated);
+        // mapper.root().updateFieldType();
```

Comment:
why it is commented, you need to update the mapper object otherwise underlying field types are not in sync ?

Reply:
leftover from debugging, I wasn't sure this was needed but I added it again.